### ISSUE TITLE
Does not always work with yarn workspaces / monorepos

### ISSUE THREAD

**Issue by buschtoens** [ember-cli/ember-cli-version-checker#70]
I'm running into problems with `ember-cli-version-checker` with the following setup: I have a monorepo with an engine, that uses `@ember-decorators/babel-transforms` to enable decorators. The engine uses `ember-cli-babel@7`, which `babel-transforms` checks for using `ember-cli-version-checker`. This works for the engine itself, however it _does not_ work for the engine's dummy app.

Here's why: the dummy app is an instance of `EmberAddon` and has no `root` property. Therefore `basedir` is `undefined`.

https://github.com/ember-cli/ember-cli-version-checker/blob/8443ac7c852aff0be73e1b8687fd912afdc6a7ae/src/npm-dependency-version-checker.js#L14-L16

This means that [`require.sync`](https://github.com/browserify/resolve/blob/3170ce1a3d3a2a14550a4f52b2d201c90d574ce7/lib/sync.js#L39) determines the `basedir` using the physical location of the caller file, `npm-dependency-version-checker.js` in that case:

```js
var basedir = opts.basedir || path.dirname(caller());
```

And this is where it breaks. `ember-cli-version-checker` has been hoisted to the workspace root, consequently `resolve` will start the lookup from there. However, the version of `ember-cli-babel@7` that is used by the engine _has not_ been hoisted. Instead, `ember-cli-babel@6` has been hoisted.

Related to #47.

**Comment by buschtoens** [comment]
I am no ember-cli wizard, but I would guess, that we could just do this:

```js
jsonPath = resolve.sync(this.name + '/package.json', { 
  basedir: addon.root || addon.project && addon.project.root, 
}); 
```

**Comment by ef4** [comment]
I just hit this bug. It gets reproduced if you try to use ember-decorators when ember-cli-babel needs to be resolved relative to a non-default basedir. This is the VersionChecker that fails:

https://github.com/ember-decorators/ember-decorators/blob/d96c7c86a36f242a2c8b52d3add333baeeeab947/packages/babel-transforms/index.js#L20

**Comment by alexlafroscia** [comment]
In case anyone comes across this in the future: `@ember-decorators/babel-transforms` just released `5.1.2` that bumps the `ember-cli-version-checker` version and (should) fix this issue if you're using that package.
